[AIE2P] Fix modeling of resource conflicts in the scheduling model

katerynamuts opened the pull request:
I've run QoR tests with the updated scheduling model and the failed and passed tests remain the same.

katerynamuts commented:
I added a test for the known conflict that is fixed by this PR, namely, the conflict between `VLDA_FILL_512 ` and `VMOV`, in the scheduling model there is a new resource called `LD_FIFO_WA_PORT` that  models the conflict.

martien-de-jong commented on llvm/test/CodeGen/AIE/aie2p/schedule/vlda_fill_512_vmov_conflict.mir:
```diff
@@ -0,0 +1,30 @@
+#
+# (c) Copyright 2025 Advanced Micro Devices, Inc. or its affiliates
+# RUN: llc -march=aie2p -run-pass=postmisched %s -o - | FileCheck %s
+
```
Perhaps add a comment about which resource we are checking and in which cycles they are used by the instruction.

Also see AIE/aie2/schedule/resource/*

Is this series of NOPs really caused by a resource conflict, or is there a data dependence?

katerynamuts replied:
I changed the tests a little bit so that there is no data dependency.

andcarminati commented on llvm/test/CodeGen/AIE/aie2p/schedule/vlda_fill_512_vmov_conflict.mir:
```diff
@@ -0,0 +1,30 @@
+    ; CHECK-NEXT: $lfl0 = VMOV_alu_mv_mv_x killed $x9
+    ; CHECK-NEXT: NOP
+    $p1, $lf0, $r25 = VLDA_FILL_512 $p1, $lf0, $r25
+    $lfl0 = VMOV_alu_mv_mv_x $x9
```
maybe `$lfl1` here to avoid any WAW dependency?

khallouh commented on llvm/test/CodeGen/AIE/aie2p/schedule/resource/ld_fifo_wa_port.mir:
```diff
@@ -0,0 +1,31 @@
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: $lfl1 = VMOV_alu_mv_mv_x killed $x9
+    ; CHECK-NEXT: NOP
+    ; CHECK-NEXT: NOP
```
Nit: maybe add the test in a pre-commit to show that the instruction is moved up by one cycle to avoid the conflict after the change.

khallouh reviewed (approved):
LGTM

katerynamuts committed:
[AIE2P] Fix modeling of resource conflicts in the scheduling model

katerynamuts committed:
[AIE2P] Add test for LD_FIFO_WA_PORT in the scheduling model.

The port is missing in the scheduling model, it is added in the next
commit. The test is added to show how the resource changes the
scheduling.
